### ISSUE TITLE
Enums with overriden methods.

### ISSUE THREAD

**Issue by daragallagher** [kostaskougios/cloning#52]
Line 414 of Cloner contains the line "if (clz.isEnum()) return o" which fails if the Enum values overload methods (a gotcha with Class.isEnum()).  As a result a new instance of the Enum value is created using objenesis which will fail to equal the original value.  The test should use "clz instanceof Enum<?>" which would catch these cases.

**Comment by kostaskougios** [comment]
do you have an example enum (or even better a testcase which fails)?

**Comment by daragallagher** [comment]
Sure, let me put a test case together.  I'll get something to you over the weekend.

-- Cheers! Dara.
________________________________

**Comment by daragallagher** [comment]
```java
import org.junit.Assert;
import org.junit.Test;

import com.rits.cloning.Cloner;

public class CloneEnumTest {

    enum Foo {
        One {
            public int getValue() {
                return 1;
            }
        },
        Two {
            public int getValue() {
                return 2;
            }
        };
        abstract int getValue();
    }
    
    @Test
    public void test() {
        Foo one = Foo.One;
        Cloner cloner = new Cloner();
        Foo clone = cloner.deepClone(one);
        Assert.assertEquals(one, clone);
    }
}
```

**Comment by kostaskougios** [comment]
Hi,

Your test passes. Not only one and cloner are equals but they are the 
same. That's the correct behavior, isn't it?

I think you're using an older version of cloner. The line you mentioned 
currently has this code:

if (clonedPreviously !=null)return (T) clonedPreviously;

What version are you using?

**Comment by daragallagher** [comment]
Sorry for wasting your time.  I was inadvertently using a slightly older version because of a transitive maven dependency.  This has been fixed in version 1.9.1; the line `if (clz.isEnum()) return o;` in 1.9.0 was changed to `if (o instanceof Enum) return o;` which correctly identifies enums like the one above.
